pkg/rpmmd/package.go

```diff
@@ -0,0 +1,172 @@
+// - get_debuginfo_name_of_source() - RPM package debuginfo package name for the source package
+// - get_debuginfo_name() - RPM package debuginfo package name
+// - get_prereq_ignoreinst()
+// - get_depends() - RPM package dependencies (requires + enhances + suggests + supplements + recommends)
+// - get_changelogs()
+// - get_hdr_end()
+// - get_media_number()
+// - get_package_path() - Path to the RPM package on the local file system
+// - is_available_locally()
+// - is_installed()
+// - is_excluded()
+// - get_from_repo_id() -  For an installed package, return id of repo from the package was installed
+// - get_install_time()
+// - get_rpmdbid()
+type Package struct {
+	Name    string
+	Epoch   uint
+	Version string
+	Release string
+	Arch    string
+
+	// RPM package Group
+	Group string
+
+	// File size of the RPM package
+	DownloadSize uint64
+	// Size the RPM package should occupy after installing on disk
+	// NB: The actual size on disk may vary based on block size and filesystem overhead.
+	InstallSize uint64
+
+	License string
+
+	// RPM package source package filename
+	SourceRpm string
+
+	BuildTime time.Time
+	Packager  string
+	Vendor    string
+
+	// RPM package URL (project home address)
+	URL string
+
+	Summary     string
+	Description string
+
+	// Regular dependencies
+	Provides        RelDepList
+	Requires        RelDepList // RegularRequires + PreRequires
+	RequiresPre     RelDepList
+	Conflicts       RelDepList
+	Obsoletes       RelDepList
+	RegularRequires RelDepList
+
+	// Weak dependencies
+	Recommends  RelDepList
+	Suggests    RelDepList
+	Enhances    RelDepList
+	Supplements RelDepList
+
+	// List of files and directories the RPM package contains
+	Files []string
+
+	// Repodata
+	// RPM package baseurl from repodata
+	BaseURL string
+	// RPM package relative path/location from repodata
+	Location string
+	// RPM package remote location where the package can be download from
+	RemoteLocations []string
+
+	// Checksum object representing RPM package checksum and its type
+	Checksum Checksum
+	// Checksum object representing RPM package header checksum and its type.
+	HeaderChecksum Checksum
+
+	// Repository ID this package belongs to
+	// XXX: We should should eventually hold a reference to the RepoConfig
+	RepoID string
+
+	// Resolved reason why a package was / would be installed.
+	Reason string
+
+	// Convenience values coming from the respective repository config
+	Secrets   string
+	CheckGPG  bool
+	IgnoreSSL bool
+}
+
+// GetEVRA returns the package's Epoch:Version-Release.Arch string.
+// If the package Epoch is 0, it is omitted and only Version-Release.Arch is returned.
+func (p *Package) GetEVRA() string {
+	if p.Epoch == 0 {
+		return fmt.Sprintf("%s-%s.%s", p.Version, p.Release, p.Arch)
+	}
+	return fmt.Sprintf("%d:%s-%s.%s", p.Epoch, p.Version, p.Release, p.Arch)
+}
+
+type PackageList []Package
+
+func (pl *PackageList) GetPackage(packageName string) (Package, error) {
```

Comment:
(super nitpick) Maybe just `Package()` (c.f. https://go.dev/doc/effective_go#Getters) but we don't really follow this in many places so feel free to ignore.

Reply:
Thanks for pointing this out, I'll adjust the getter, as well as other getters for the `rpmmd.Package` struct. Shorter names and following Go idioms is a good thing :wink: